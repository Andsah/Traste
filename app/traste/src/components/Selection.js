import React from 'react';
import PropTypes from 'prop-types';
import {TextField, MenuItem} from '@mui/material';

/**
 * Custom TextField for displaying a selectionbox with different alternatives.
 * @param {*} label Name of field.
 * @param {*} value Starting value.
 * @param {*} onChange Handler.
 * @param {*} error What will be shown for the error.
 * @param {*} data A list of data to be displayed.
 * @return {TextField} Rendred field.
 */
function Selection({label, value, onChange, error, data}) {
  return (
    <TextField
      select
      label={label}
      value={value}
      onChange={onChange}
      sx={{
        marginTop: '15px',
        backgroundColor: 'rgba(255,255,255,0.3)',
        width: '90vw',
      }}
      // required
      inputProps={{'data-testid': 'selectionfield'}}
      error={!!error}
      // helperText={error ? error.message : null}
    >
      {
        /** Placeholde ifall det inte finns nÃ¥gon data */
        data !== undefined ? (
          data.map((option) => (
            <MenuItem key={option.id} value={option.label}>
              {option.label}
            </MenuItem>
          ))
        ) : (
          <MenuItem key={0}>Ingen data</MenuItem>
        )
      }
    </TextField>
  );
}

Selection.propTypes = {
  label: PropTypes.func.isRequired,
  value: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  error: PropTypes.func.isRequired,
  data: PropTypes.func.isRequired,
};

export default Selection;
