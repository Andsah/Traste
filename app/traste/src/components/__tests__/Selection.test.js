import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import Selection from "../Selection";

afterEach(() => {
  cleanup();
});

test("Should render Selection with selection length of 4", () => {
  const propsData = {
    title: "Bin Size",
    name: "BinSize",
    data: [
      {
        id: "0",
        label: "5",
      },
      {
        id: "1",
        label: "10",
      },
      {
        id: "2",
        label: "15",
      },
      {
        id: "3",
        label: "20",
      },
    ],
  };

  render(
    <Selection
      title={propsData.title}
      name={propsData.name}
      data={propsData.data}
    />
  );
  const selectionElement = screen.getByTestId("selectionfield");
  expect(selectionElement).toBeInTheDocument();
  //expect(selectionElement).toHaveLength(propsData.length);
  expect(selectionElement).toHaveAttribute("name",propsData.name)
});

test("Matches snapshot", () => {
  const tree = renderer.create(<Selection />).toJSON();
  expect(tree).toMatchSnapshot();
});
