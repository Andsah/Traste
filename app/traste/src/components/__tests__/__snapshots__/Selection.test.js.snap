// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-1hg5lel-MuiFormControl-root-MuiTextField-root"
>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
    variant="outlined"
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="select-undefined"
      className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="select-undefined"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="notranslate"
      >
        ​
      </span>
    </div>
    <input
      aria-hidden={true}
      autoFocus={false}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      data-testid="selectionfield"
      disabled={false}
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value=""
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-nnbavb"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
